base_layer/key_manager/src/cipher_seed.rs

```diff
@@ -113,12 +116,13 @@ pub const CIPHER_SEED_MAC_BYTES: usize = 5;
 pub struct CipherSeed {
     version: u8,
     birthday: u16,
-    entropy: [u8; CIPHER_SEED_ENTROPY_BYTES],
-    salt: [u8; CIPHER_SEED_SALT_BYTES],
+    entropy: Vec<u8>,
```

Comment:
Why should we replace fixed size arrays to `Vec`'s, in this case ? I personally feel it is expressive to know in advance the size of both `entropy` and `salt`.

Reply:
Using a `Vec` made it easier to handle zeroizing of secret data (due to heap storage), and also should keep the structure more general in case of future updates, where data sizes could change based on version. I could easily imagine a future where all size constants are part of a version-dependent lookup.

Reply:
I'm certainly open to changing back to arrays for now, but it seems like this requires much more careful zeroizing wrappers to account for copies/moves, which seems like a footgun to avoid.

Reply:
`Box<[u8; CIPHER_SEED_SALT_BYTES]>` is also an option as that allocates a fixed size array on the heap and you move a pointer (Box) around, not the data. But happy to keep it as a vec

Reply:
Thanks for the explanations @AaronFeickert. I would say that @sdbondi suggestion is the most explicit. I am also happy to use `Vec` here, it is probably not worth spending time on refactoring it it, as of now.

Reply:
The refactoring was pretty straightforward, so I went ahead and switched from `Vec<u8>` to `Box<[u8]>` for clarity on sizes. For some reason `Zeroizing` didn't like `Box<[u8]>`, so there's one spot where we do `Zeroizing<[u8]>` and then create the `Box` later; it didn't complain when we `#[derive(Zeroize)]` on `Box<[u8]>` though, which is odd...